RacingCar/main.swift

```diff
@@ -16,3 +16,108 @@ final class StringControl {
             .replacingOccurrences(of: ")", with: "")
     }
 }
+
+extension String {
+    
+    enum ArithmeticalCode: String {
+        case add = "+"
+        case division = "/"
+        case multiply = "*"
+        case subtract = "-"
+    }
+    
+    func isArithmeticalCode() -> Bool {
+        return ArithmeticalCode(rawValue: self) != nil ? true : false
+    }
+    
+    func toArithmeticalCode() -> ArithmeticalCode? {
+        return ArithmeticalCode(rawValue: self)
+    }
+}
+
+struct StringCalculator {
+    
+    enum CalculatorError: Error {
+        case stringIsEmptyOrNil
+        case containsWrongOperatorCode
+    }
+    
+    typealias ArithmeticalExpression = (Int, Int) -> Int
+    
+    func containsWrongArithmeticalCode(string arithmeticString: String) -> Bool {
+        let wrongCodes = arithmeticString
+            .components(separatedBy: " ")
+            .filter{ Int($0) == nil }
+            .filter { $0.isArithmeticalCode() == false }
+        
+        return wrongCodes.isEmpty == false
+    }
+    
+    func calculate(string arithmeticString: String?) throws -> Int? {
+        guard let arithmeticString = arithmeticString,
+              arithmeticString.components(separatedBy: " ").isEmpty == false
+        else {
+            throw CalculatorError.stringIsEmptyOrNil
+        }
+        
+        guard containsWrongArithmeticalCode(string: arithmeticString) == false else {
+            throw CalculatorError.containsWrongOperatorCode
+        }
+        
+        var result: Int = 0
+        let slicedString = arithmeticString.components(separatedBy: " ")
+        
+        if let firstNum = Int(slicedString.first!) {
```

Comment:
아무리 간단한 변수이름이라도 최대한 축약을 지양합니다.

Reply:
넵 좀 더 알기 쉽게 수정해보겠습니다.

Reply:
축약을 지양한다는 내용의 [Swift API Design Guidelines](https://www.swift.org/documentation/api-design-guidelines/#use-terminology-well) 문서를 참고차 첨부합니다~

> Avoid abbreviations. Abbreviations, especially non-standard ones, are effectively terms-of-art, because understanding depends on correctly translating them into their non-abbreviated forms.